worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StoragePolicy.scala

```diff
@@ -140,7 +140,8 @@ class StoragePolicy(conf: CelebornConf, storageManager: StorageManager, source:
                 partitionDataWriterContext.isPartitionSplitEnabled)
               partitionDataWriterContext.setWorkingDir(workingDir)
               val metaHandler = getPartitionMetaHandler(diskFileInfo)
-              if ((storageInfoType == StorageInfo.Type.HDD || storageInfoType == StorageInfo.Type.SSD) && location.getStorageInfo.localDiskAvailable()) {
+              val diskFull = checkDiskFull(flusher)
+              if ((storageInfoType == StorageInfo.Type.HDD || storageInfoType == StorageInfo.Type.SSD) && location.getStorageInfo.localDiskAvailable() && !diskFull) {
```

Comment:
Could we add this check into `StorageManager#createDiskFile` instead of before creation of `LocalTierWriter`?